packages/editor/src/components/header/index.js

```diff
@@ -32,6 +32,9 @@ import {
 } from '../../store/constants';
 import { unlock } from '../../lock-unlock';
 
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
```

Comment:
Maybe we should also add a check for `globalThis.IS_GUTENBERG_PLUGIN` to tree-share the code for the WP core.

Reply:
Will this be addressed in a follow-up? Or in this PR, I think that's fine.

Reply:
I'm not 100% sure if it's needed. Looking at other experiments, none of them use `globalThis.IS_GUTENBERG_PLUGIN`.

Reply:
My guess is that `globalThis.IS_GUTENBERG_PLUGIN` prevents the experimental code itself from being shipped into core, but I'm not sure. In any case, I think it's okay to ship this PR for now